foundations/src/telemetry/tracing/output_otlp_grpc.rs

```diff
@@ -62,19 +63,22 @@ async fn do_export(
             .map(|span| convert_span(span, &service_info))
             .collect();
 
-        client
-            .ready()
-            .await
-            .context("lost gRPC connection to the trace service")?;
+        if let Err(err) = client.ready().await {
+            reporter_error(err);
+            continue;
+        }
 
-        client
+        let send_res = client
             .unary::<_, ExportTraceServiceResponse, _>(
                 create_request(resource_spans),
                 PathAndQuery::from_static(COLLECTOR_PATH),
                 ProstCodec::default(),
             )
-            .await
-            .context("failed to send gRPC request to the trace service")?;
+            .await;
+
+        if let Err(err) = send_res {
+            reporter_error(err);
+        }
```

Comment:
Can't comment on an unchanged line --- what? github lol

But should this method `do_export` not return `BootstrapResult` at all now?